components/home/reducers.js

```diff
@@ -1,34 +1,43 @@
+import uuid from 'uuid/v1';
+
 import { UPDATE_REACTION, REQUEST_UPLOAD_LIST } from './actionTypes';
-import { INITIAL_STATE, USER_UPLOADS } from './constants';
+import { INITIAL_STATE } from './constants';
 
-/**
- * Creates a Javascript Map with the user uploads mapped by id
- *
- * @param {Array} USER_UPLOADS - a users uploads
- * @return {Map} - the user uploads
- */
+import fakeUploads from '../../models/fakeUploads';
 
+/**
+* Creates a Javascript Map with the user uploads mapped by id
+*
+* @param {Array} fakeUploads - a users uploads
+* @return {Map} - the user uploads
+*/
 function generateUploadsMap() {
     const uploads = new Map();
 
-    USER_UPLOADS.forEach(userUpload => {
-        const { id } = userUpload;
-
-        uploads.set(id, userUpload);
+    fakeUploads.forEach(upload => {
+        const { id } = upload;
+        uploads.set(id, upload);
     });
 
     return uploads;
 }
 
 function updateUploadReaction(id, type, uploads) {
     const updatedUploads = new Map([...uploads.entries()]);
-    const userUpload = updatedUploads.get(id);
-    if (!userUpload.reactions[type]) {
-        userUpload.reactions[type] += 1;
-    } else {
-        userUpload.reactions[type] -= 1;
-    }
-    updatedUploads.set(id, userUpload);
+    const upload = updatedUploads.get(id);
+    const userId = uuid();
+
+    uploads.forEach(userUpload => {
+        const { users } = userUpload.reactions[type];
+        if (Object.values(users).includes(userId[userId])) {
```

Comment:
what is `userId[userId]`?

Reply:
It is supposed to be the value of UserId.